VersionControlUI/src/DiffManager.cpp

```diff
@@ -0,0 +1,300 @@
+
+		if (!oldNode)
+			continue;
+
+		 Visualization::VisualizationManager::instance().mainScene()->addPostEventAction(
+								[oldNode, diffViewItem](){
+			 auto oldItems =  diffViewItem->findAllVisualizationsOf(oldNode);
+			 for (auto item : oldItems)
+			 {
+				 auto overlay = new Visualization::HighlightOverlay{item,
+						 Visualization::HighlightOverlay::itemStyles().get("delete_no_bg_solid")};
+				 item->addOverlay(overlay, "delete_highlights");
+			 }});
+
+	}
+
+	// create highlights for added nodes
+	for (auto id : nodeLists[INDEX_OF_ADD])
+	{
+		auto newNode = const_cast<Model::Node*>(newVersionManager->nodeIdMap().node(id));
+
+		Visualization::VisualizationManager::instance().mainScene()->addPostEventAction(
+							  [newNode, diffViewItem](){
+			auto newItems =  diffViewItem->findAllVisualizationsOf(newNode);
+			for (auto item : newItems)
+			{
+				auto overlay = new Visualization::HighlightOverlay{item,
+						Visualization::HighlightOverlay::itemStyles().get("insert_no_bg_solid")};
+				item->addOverlay(overlay, "insert_highlights");
+			}});
+	}
+
+	// create highlights for modified nodes
+	for (auto id : nodeLists[INDEX_OF_MODIFY])
+	{
+		auto oldNode = const_cast<Model::Node*>(oldVersionManager->nodeIdMap().node(id));
+		auto newNode = const_cast<Model::Node*>(newVersionManager->nodeIdMap().node(id));
+
+		Visualization::VisualizationManager::instance().mainScene()->addPostEventAction(
+							  [oldNode, newNode, diffViewItem](){
+			auto items = diffViewItem->findAllVisualizationsOf(oldNode);
+			items.append(diffViewItem->findAllVisualizationsOf(newNode));
+			for (auto item : items)
+			{
+				auto overlay = new Visualization::HighlightOverlay{item,
+						Visualization::HighlightOverlay::itemStyles().get("modify_no_bg_solid")};
+				item->addOverlay(overlay, "modify_highlights");
+			}});
+	}
+
+	// switch to the newly created view
+	Visualization::VisualizationManager::instance().mainScene()->viewItems()->switchToView(diffViewItem);
+}
+
+/**
+ * Deletes all nodes in \a container if their direct parent is also in \container.
+ */
+template<typename Container>
+void DiffManager::removeDirectChildrenOfNodesInContainer(Container* container)
+{
+	auto it = container->begin();
+	while (it != container->end())
+	{
+		auto nodeInContainer = Model::AllTreeManagers::instance().nodeForId(*it);
+		auto parentNode = nodeInContainer->parent();
+
+		if (container->contains(Model::AllTreeManagers::instance().idForNode(parentNode)))
+			it = container->erase(it);
+		else
+			it++;
+	}
+}
+
+// TODO change method to use type<> to specify ancestor
+/**
+ * Searches in \a treeManager for a node with \a id and tries to get it's first ancestor of type OOModel::Class.
+ * Returns true if such a node was found and saved to \a resultId.
+ */
+bool DiffManager::findChangedNode(Model::TreeManager* treeManager, Model::NodeIdType id, Model::NodeIdType& resultId)
```

Comment:
Instead of returning the Id, can't you directly return the node itself? That way you can return nullptr if nothing was found, and you don't need the last argument.

Is there some reason why returning the ID is better?